### ISSUE TITLE
SimpleITK 2.1.1.2 causes a segfault when reading a truncated JPEG file from scikit-image

### ISSUE THREAD

**Issue by grlee77** [SimpleITK/SimpleITK#1631]
At scikit-image, we have observed CI failures (which I am also able to reproduce locally) when running our test suite with SimpleITK 2.1.1.2. The specific failing test case involves reading in a truncated JPEG file and is expected to produce a `RuntimeError`.

A minimal example to reproducing the segfault outside of the test suite is
```Python
from skimage import io
from skimage._shared import testing

io.use_plugin('simpleitk')
truncated_jpg_filename = testing.fetch('data/truncated.jpg')
io.imread(truncated_jpg_filename)
```

Links to the CI logs with the segfaults are given at https://github.com/scikit-image/scikit-image/issues/6349. relevant section of the traceback:
```
Current thread 0x00007fdd7a93e740 (most recent call first):
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/SimpleITK/SimpleITK.py", line 8015 in Execute
  File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/SimpleITK/extra.py", line 346 in ReadImage
  File "/home/runner/work/scikit-image/scikit-image/skimage/io/_plugins/simpleitk_plugin.py", line 15 in imread
  File "/home/runner/work/scikit-image/scikit-image/skimage/io/manage_plugins.py", line 207 in call_plugin
  File "/home/runner/work/scikit-image/scikit-image/skimage/io/_io.py", line 53 in imread
  File "/home/runner/work/scikit-image/scikit-image/skimage/io/tests/test_simpleitk.py", line 48 in test_imread_truncated_jpg
```

We do not see this segfault with prior versions of SimpleITK or when testing locally with 2.2.0rc2 or the latest development binary.

**Comment by blowekamp** [comment]
This reported issue appears to be with with Python 3.8 with the manylinux 2014 x86_64 wheel. Is the issue reproducible on other platforms or wheels?

**Comment by grlee77** [comment]
I also see it locally on a linux system with either Python 3.9 or 3.10. The specific wheels used are:
`SimpleITK-2.1.1.2-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl`
`SimpleITK-2.1.1.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl`

**Comment by grlee77** [comment]
have not tried on other platforms yet

**Comment by blowekamp** [comment]
On my intel Mac with Python 3.8, I got the following:
```
>>> io.imread(truncated_jpg_filename)
Premature end of JPEG file
Bogus Huffman table definition
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/blowekamp/miniconda3/envs/skimage/lib/python3.8/site-packages/skimage/io/_io.py", line 53, in imread
    img = call_plugin('imread', fname, plugin=plugin, **plugin_args)
  File "/Users/blowekamp/miniconda3/envs/skimage/lib/python3.8/site-packages/skimage/io/manage_plugins.py", line 207, in call_plugin
    return func(*args, **kwargs)
  File "/Users/blowekamp/miniconda3/envs/skimage/lib/python3.8/site-packages/skimage/io/_plugins/simpleitk_plugin.py", line 15, in imread
    sitk_img = sitk.ReadImage(fname)
  File "/Users/blowekamp/miniconda3/envs/skimage/lib/python3.8/site-packages/SimpleITK/extra.py", line 346, in ReadImage
    return reader.Execute()
  File "/Users/blowekamp/miniconda3/envs/skimage/lib/python3.8/site-packages/SimpleITK/SimpleITK.py", line 8015, in Execute
    return _SimpleITK.ImageFileReader_Execute(self)
RuntimeError: Exception thrown in SimpleITK ImageFileReader_Execute: /Users/runner/work/1/sitk-build/ITK/Modules/IO/JPEG/src/itkJPEGImageIO.cxx:309:
ITK ERROR: JPEGImageIO(0x7f970e219e30): Error JPEGImageIO could not open file: /Users/blowekamp/Library/Caches/scikit-image/0.19.2/data/truncated.jpg
```

I am assuming that is how the test is suppose to fail.

**Comment by blowekamp** [comment]
I was able to reproduce the segmentation fault on Linux too with SimpleITK-2.1.1.2-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl. Interestingly when I tried the older manylinux 2010 wheel: SimpleITK-2.1.1.2-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl there was no segmentation fault.

I am suspicious that there is some conflicting C++ symbol similar to the issues linked in this change:
https://github.com/SimpleITK/SimpleITK/commit/ada61a499c68c41999cb442b5e79c328410e2d47

**Comment by blowekamp** [comment]
The manylinux2014 wheels were not in 2.1.1 but adding with 2.1.1.2.

The manylinux 2014 wheel work in 2.2rc2 and the current nightly 2.2.0rc2.post68 wheels for x86_64 python 3.8. However, this above test still seg faults for 2.2rc1.

Can you please confirm that the manylinux2010 wheels work? 
We are considering yanking or deleting just the manylinux2014 wheels from 2.1.1.2 while leaving the manylinux2010 wheels.

**Comment by grlee77** [comment]
Thanks, I can confirm that the `SimpleITK-2.1.1.2-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl` works fine in local testing. I guess if you pull the manylinux2014 wheels then we won't even have to worry about updating the pinning in scikit-image. Thanks for the quick response on this!

**Comment by blowekamp** [comment]
The manylinux2014 wheels have been yanked. Which seems to resolve the failing test.

The underlying issue was in ITK <v5.3. It was reported in InsightSoftwareConsortium/ITK#2928 and fixed in PR InsightSoftwareConsortium/ITK#2933. We are anxiously awaiting using ITK v5.3.0 in SimpleITK 2.2 to contain this fix.

**Comment by grlee77** [comment]
Thanks, I will go ahead and close as the broken wheels were removed and a fix has already been made for ITK 5.3.
